projects/vdk-core/src/vdk/api/plugin/plugin_input.py

```diff
@@ -198,6 +202,33 @@ def ingest_payload(
             invocations belong to same collection. Defaults to "data_job_name|OpID",
             meaning all method invocations from a data job run will belong to the
             same collection.
+
+        :return: [Optional] IngestionResult, containing plugin-defined
```

Comment:
A possible use-case could be if a plugin developer decides that their ingestion mechanism returns some data, e.g., after a request is made to an ingestion endpoint, the said endpoint may return information like status code, an ingestion id, etc., and this data may, then be subject of further processing as telemetry.
Does the example make sense?

Reply:
I am not aware of such an use case, for ingestion to return metadata on data ingestion itself.
Telemetry-wise, I believe we are interested if data was ingested or not, if not eventually what the issue was.

There is no explicit pydoc on ingest_payload() in case of ingestion error, yet I would presume an exception is raised (evaluated in try block).

Reply:
Well, different ingestion plugins may have different ways of handling ingestion. For example, a plugin may calculate payload size (compressed, uncompressed) before ingesting the data, or do some other calculation. Additionally, an ingestion plugin may do its own exception handling, and passing the information about the exceptions as an IngestionResult would allow for more granular data gathering.

Reply:
Got it, thank you for elaborating